app/src/org/commcare/android/view/GraphView.java

```diff
@@ -103,6 +112,9 @@ public Intent getIntent(GraphData data) throws InvalidStateException {
         if (mData.getType().equals(Graph.TYPE_TIME)) {
             return ChartFactory.getTimeChartIntent(mContext, mDataset, mRenderer, title, getTimeFormat());
         }
+        if (mData.getType().equals(Graph.TYPE_BAR)) {

```

Comment:
Might be a good idea to flip this to Graph.TYPE_BAR.equals(mData.getType()) in order to avoid null pointer exceptions.